.docs/content/1.concepts/10.plugins.md

```diff
@@ -0,0 +1,47 @@
+# Using external plugins with ArmoniK.Core
+
+## Queue adaptors as external plugins
+
+We have the possibility to have a queue adaptor as an external project and to use it as a plugin to ArmoniK.Core when needed. More about adaptors is available [here](https://github.com/aneoconsulting/ArmoniK.Core/blob/main/.docs/content/1.concepts/7.adaptors.md). Using a queue adaptor as an external plugin was possible thanks to dynamic loading of adaptors, and the use of docker images.
+
+### Loading queue adaptor into a C# project
+
+ArmoniK.Core is based on C# projects where queue adaptors are dynamically loaded. Dynamic assembly loading allows to create dotnet applications with plugins. We will refer to the application to which the plugins will be connected as the hosting application.
+
+The hosting application should define an interface that all plugins need to implement. This interface serves as a base for each plugin. In our case, the base interface that should be implemented in any plugin is in NuGet package [ArmoniK.Core.Base](https://www.nuget.org/packages/ArmoniK.Core.Base/). Any plugin should have a reference to this package.
+
+### Packing a plugin
+
+Once the queue adaptor is implemented, we need to pack it to be loaded in ArmoniK.Core. To pack the plugin to ArmoniK.Core, we modify the docker images of the polling agent and the control plane. We should create two images: one based on a specific version of the polling agent, with the plugin project added on top, and the other based on a specific version of the control plane image, also with the plugin project added on top.
```

Comment:
Once the queue adaptor is implemented, we need to pack it to be loaded in ArmoniK.Core. 

To pack the plugin to ArmoniK.Core, we need to modify the docker images of the polling agent and the submitter. We should create two images: 
- one based on a defined version of the polling agent, with the plugin project added on top, 
- one based on a defined version of the control plane, with the plugin project added on top.